doc/source/reference/c-api.dtype.rst

```diff
@@ -374,3 +374,22 @@ format specifier in printf and related commands.
     :c:data:`NPY_LONGLONG_FMT`, :c:data:`NPY_ULONGLONG_FMT`,
     :c:data:`NPY_INTP_FMT`, :c:data:`NPY_UINTP_FMT`,
     :c:data:`NPY_LONGDOUBLE_FMT`
+
+Generic Typing and Deviations from C Standard Syntax
+----------------------------------------------------
+
+Many of the C source files in NumPy are not compliant with standard
```

Comment:
would be good to make clear that this is about `.c.src` files. The current sentence suggest that we have `.c` files that aren't C89/99 compliant.